client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -1070,6 +1077,25 @@
                     }
                 };
 
+                scope.canSendAndContinue = function() {
+                    return !authoring.isPublished(scope.item);
+                };
+
+                scope.sendAndContinue = function() {
+                    var deskId = scope.selectedDesk._id;
+                    var stageId = scope.selectedStage._id || scope.selectedDesk.incoming_stage;
+                    var activeDeskId = desks.activeDeskId;
+                    scope.item.more_coming = true;
+                    return sendAuthoring(deskId, stageId, scope.selectedMacro, true)
+                        .then(function() {
+                            authoring.linkItem(scope.item, null, activeDeskId).then(function (item) {
```

Comment:
IMO, flattening the promise chain is a bit more readable than having nested `.then()` blocks.